### PR TITLE
Storage queue from Alan's PR

### PR DESCRIPTION
**Pull request by sima-zhu**
- [x] [Code](https://github.com/alzimmermsft/azure-sdk-for-java/blob/AzStorage_Queues_Merged_Clients/storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java#L135), Mono<Response> -> Flux<DM>

- [x] Whether merge in DequedMessage and PeekedMessage? No merge. Confirmed from Alan.

- [x] [Code](https://github.com/alzimmermsft/azure-sdk-for-java/blob/AzStorage_Queues_Merged_Clients/storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java#L77), getURL() or others? File an issue

- [x] Why 'Azure' in AzureQueueStorageBuilder? Already renamed in code.

- [x] connectionString() or credential() in builder patter

- [ ] [Code](https://github.com/alzimmermsft/azure-sdk-for-java/blob/AzStorage_Queues_Merged_Clients/storage/client/src/main/java/com/azure/storage/queue/models/DequeuedMessage.java#L17) Is there ever a time a user would want to mutate or instantiate this class? To me it would be reasonable to make this an immutable class? It is from autogeneration. Will sync with Jianghao after blob change.

- [x] [Code](https://github.com/alzimmermsft/azure-sdk-for-java/blob/AzStorage_Queues_Merged_Clients/storage/client/src/main/java/com/azure/storage/queue/models/QueueProperties.java#L29) POJO or Fluent. Will do with POJO

- [x] [Code](https://github.com/alzimmermsft/azure-sdk-for-java/blob/AzStorage_Queues_Merged_Clients/storage/client/src/main/java/com/azure/storage/queue/models/UpdatedMessage.java#L28) This class API and the name of the class are not obviously related. Is UpdatedMessage really the right name here? This data model is used in QueueAsyncClient for updateMessage API, so the name is by design.

### CODE REVIEW

**Comment by sima-zhu** [comment]
Two undone in checklist. 
One will follow up with Jianghao.
One will address in this PR.
Other checklist please refer the comments in the list.

**Comment by sima-zhu** [comment]
Need to have code snippet in next PR.

File: pom.client.xml [hunk 1]
```diff
@@ -523,6 +523,7 @@
                 -snippetpath ${project.basedir}/eventhubs/client/azure-eventhubs/src/samples/java
                 -snippetpath ${project.basedir}/keyvault/client/keys/src/samples/java
                 -snippetpath ${project.basedir}/keyvault/client/secrets/src/samples/java
+                -snippetpath ${project.basedir}/storage/client/queue/src/samples/java/javadoc
```
Comment (JonathanGiles) [inline pom.client.xml]: There shouldn't be `javadoc` at the end of this line

File: storage/client/pom.xml [hunk 2]
```diff
@@ -115,6 +114,23 @@
...
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>com.github.spotbugs</groupId>
+        <artifactId>spotbugs-maven-plugin</artifactId>
+        <configuration>
+          <failOnError>false</failOnError>
+        </configuration>
+      </plugin>
```
Comment (JonathanGiles) [inline storage/client/pom.xml]: These changes can all be removed - it is centrally managed now.

File: storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java [hunk 3]
```diff
@@ -1,29 +1,535 @@
...
+ *
+ * <p><strong>Instantiating an Asynchronous Queue Client</strong></p>
+ *
+ * <pre>
+ * QueueAsyncClient client = QueueAsyncClient.builder()
+ *     .connectionString(connectionString)
+ *     .endpoint(endpoint)
+ *     .queueName(queueName)
+ *     .buildAsync();
+ * </pre>
```
Comment (JonathanGiles) [inline storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java]: This code snippet should be in an external .java file like other code snippets.

File: storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java [hunk 4]
```diff
@@ -1,29 +1,535 @@
...
+     * Creates a new queue.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Create a queue with metadata "queue:metadataMap"</p>
+     *
+     * <pre>
+     * client.create(Collections.singletonMap("queue", "metadataMap"))
+     *     .subscribe(response -&gt; System.out.printf("Create completed with status code %d", response.statusCode()));
+     * </pre>
```
Comment (JonathanGiles) [inline storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java]: Make sure all code snippets are externalised - I won't keep mentioning this as I get further down the PR.

File: storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java [hunk 5]
```diff
@@ -1,29 +1,535 @@
...
+    }
+
+    /**
+     * Permanently deletes the queue.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Delete a queue</p>
+     *
+     * @codesnippet com.azure.storage.queue.queueAsyncClient.delete
```
Comment (JonathanGiles) [inline storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java]: None of these codesnippet blocks are valid - you are missing the opening and closing brace. Generate the JavaDoc and you'll see what I mean.

File: storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java [hunk 6]
```diff
@@ -1,29 +1,535 @@
...
+        return client.queues().deleteWithRestResponseAsync(queueName, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /**
+     * Retrieves metadata and approximate message count of the queue.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Get the properties of the queue</p>
```
Comment (JonathanGiles) [inline storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java]: Code snippet titles should be more bold than a simple paragraph block.

File: storage/client/pom.xml (hunk 2 above)
Comment (sima-zhu) [reply storage/client/pom.xml]: Checkstyle should be removed too, right?

Comment (JonathanGiles) [reply storage/client/pom.xml]: Yes, all configuration of these features is centralised

**Pull request closed without merging**